g2p/cli.py

```diff
@@ -62,103 +136,185 @@ def cli():
-        This assumes the mapping IN_LANG -> IN_LANG-ipa exists and creates a mapping
-        from the its output inventory.
+def generate_mapping(
+    in_lang, out_lang, dummy, ipa, list_dummy, out_dir, merge, from_langs, to_langs
+):
+    """ Generate a new mapping from existing mappings in the g2p system.
 
-        To generate a mapping from IN_LANG-ipa to eng-ipa from a mapping following a
-        different patterns, e.g., from crl-equiv -> crl-ipa, specify both IN_LANG
-        (crl-equiv in this example) and OUT_LANG (crl-ipa in this example).
+        This command has different modes of operation:
 
-        If you just modified or created the IN_LANG to IN_LANG-ipa/OUT_LANG mapping,
-        don't forget to call "g2p update" first so "g2p generate-mapping" sees the
-        latest version.
+        Standard mode:
 
-        Call "g2p update" again after calling "g2p generate-mapping" to make the new
-        IN_LANG-ipa/OUT_LANG to eng-ipa mapping available.
+          g2p generate-mapping [--dummy|--ipa] IN_LANG [OUT_LANG]
 
-        Note: at least one of --ipa, --dummy or --list-dummy is required.
+          For specified IN_LANG, generate a mapping from IN_LANG-ipa to eng-ipa,
+          or from IN_LANG-ipa to a dummy minimalist phone inventory.
+          This assumes the mapping IN_LANG -> IN_LANG-ipa exists and creates a mapping
+          from its output inventory.
 
-        You can list available mappings with "g2p doctor --list-ipa", or by visiting
-        http://g2p-studio.herokuapp.com/api/v1/langs .
+          To generate a mapping from IN_LANG-ipa to eng-ipa from a mapping following a
+          different patterns, e.g., from crl-equiv -> crl-ipa, specify both IN_LANG
+          (crl-equiv in this example) and OUT_LANG (crl-ipa in this example).
 
-        \b
-        Sample usage:
-            Generate Algonquin IPA to English IPA from alq -> alq-ipa
+          \b
+          Sample usage:
+            Generate Algonquin IPA to English IPA from alq -> alq-ipa:
                 g2p generate-mapping --ipa alq
-            Generate Mohawk IPA to English IPA from moh-equiv -> moh-ipa
+            Generate Mohawk IPA to English IPA from moh-equiv -> moh-ipa:
                 g2p generate-mapping --ipa moh-equiv moh-ipa
             Generate Michif IPA to English IPA from the union of crg-dv ->
-            crg-ipa and crg-tmd -> crg-ipa
+            crg-ipa and crg-tmd -> crg-ipa:
                 g2p generate-mapping --ipa --merge crg-dv:crg-tmd crg-ipa
-    """
 
-    if merge:
-        if out_lang is None:
-            raise click.BadParameter("OUT_LANG is required with --merge.")
-        in_langs = in_lang.split(":")
-    else:
-        in_langs = [in_lang]
+        List the dummy inventory used by --dummy:
+
+          g2p generate-mapping --list-dummy
+
+        From/to IPA mode:
+
+          g2p generate-mapping --from FROM_L1[:FROM_L2[:...]] --to TO_L1[:TO_L2[:...]]
```

Comment:
I find this line very confusing...what is with all the nested structure?

Reply:
Fixed